applications/RANSApplication/custom_conditions/vms_monolithic_k_based_wall_condition.h

```diff
@@ -0,0 +1,278 @@
+    {
+        Condition::operator=(rOther);
+        return *this;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Create a new VMSMonolithicKBasedWallCondition object.
+    /**
+      @param NewId Index of the new condition
+      @param ThisNodes An array containing the nodes of the new condition
+      @param pProperties Pointer to the element's properties
+      */
+    Condition::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<VMSMonolithicKBasedWallCondition>(
+            NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+    }
+
+    Condition::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<VMSMonolithicKBasedWallCondition>(
+            NewId, pGeom, pProperties);
+    }
+
+    /**
+     * Clones the selected element variables, creating a new one
+     * @param NewId the ID of the new element
+     * @param ThisNodes the nodes of the new element
+     * @param pProperties the properties assigned to the new element
+     * @return a Pointer to the new element
+     */
+
+    Condition::Pointer Clone(
+        IndexType NewId,
+        NodesArrayType const& rThisNodes) const override
+    {
+        Condition::Pointer p_new_condition = Create(
+            NewId, this->GetGeometry().Create(rThisNodes), this->pGetProperties());
+
+        p_new_condition->SetData(this->GetData());
+        p_new_condition->SetFlags(this->GetFlags());
+
+        return p_new_condition;
+    }
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Initialize() override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+
+protected:
+    ///@name Protected Operations
+    ///@{
+
+    /**
+     * @brief Applies rans based wall law
+     *
+     * This method calculate left hand side matrix and right hand side vector for following equation
+     *
+     * \[
+     *      u_\tau = max\left(C_\mu^0.25 \sqrt{k}, \frac{||\underline{u}||}{\frac{1}{\kappa}ln(y^+)+\beta}\right)
+     * \]
+     *
+     * integration point value = \rho \frac{u_\tau^2}{||\underline{u}||}\underline{u}
+     *
+     * @param rLocalMatrix         Left hand side matrix
+     * @param rLocalVector         Right hand side vector
+     * @param rCurrentProcessInfo  Current process info from model part
+     */
+    void ApplyWallLaw(
+        MatrixType& rLocalMatrix,
+        VectorType& rLocalVector,
+        ProcessInfo& rCurrentProcessInfo) override;
```

Comment:
```suggestion
        const ProcessInfo& rCurrentProcessInfo) override;
```

Reply:
This I cannot do, since I am overriding the base class method. :/ We have to change it in the FluidApp first.